app/src/main/java/com/raghav/spacedawn/ui/AppViewModelFactory.kt

```diff
@@ -9,7 +9,7 @@ class AppViewModelFactory(
     private val app : Application,
     private val repository: AppRepository
 ): ViewModelProvider.Factory {
-    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
```

Comment:
i would rather prefer to have this done with help of dagger

Reply:
Agree, currently the project doesn't has any DI framework implemented. Should I make another PR for adding Dagger or should I do that in this PR itself? 
Also can you attach appropriate resources for setting-up/adding Dagger to an android project?